[mautic] add guide for mautic

putzwasser opened the pull request.

noave commented:
Thanks for your work. Unfortunately I had problems reviewing it in detail in the current state, please note some general things at first:

- please check the preview at https://deploy-preview-1506--uberlab.netlify.app/guide_mautic/ and you will see some errors yourself.

----

- keep it simple, do not use a subdomain until its necessary out of specific reasons. If it is, please state that in the guide
- assume a clean new Uberspace
- the most code blocks are misaligned, please use only 2 empty spaces to indent
- please to not separate a command and the output in two different blocks
- do not use full paths in the prompt like `[isabell@stardust /var/www/virtual/isabell]$` but only the curent folder `[isabell@stardust isabell]$` because this is the standard prompt on Uberspace
- dont tell users which editor to use or automatically pipe data into files by commands. Just tell them to use their preferred editor and put a block of lines to a file (see how its done in other guides)
- take care to use double ticks `` for inline code (single ticks are displayed as italic text)

---- 

- Please mind the Style Rules: https://github.com/Uberspace/lab/blob/main/STYLE.md
- Lookup other guides for inspiration: https://lab.uberspace.de

putzwasser commented:
Thanks for your hints

> keep it simple, do not use a subdomain until its necessary out of specific reasons. If it is, please state that in the guide

It is for GDPR reasons. I made it more obvious.

> assume a clean new Uberspace

I did, but I added a symbolic link for `html`, too.

> the most code blocks are misaligned, please use only 2 empty spaces to indent

Yeah that must happened during the finishing touches. Fixed.

> please do not separate a command and the output in two different blocks

Done

> do not use full paths in the prompt like `[isabell@stardust /var/www/virtual/isabell]$` but only the curent folder `[isabell@stardust isabell]$` because this is the standard prompt on Uberspace

Done

> dont tell users which editor to use or automatically pipe data into files by commands. Just tell them to use their preferred editor and put a block of lines to a file (see how its done in other guides)

Done, I guess, even though I feel this makes things more complicated than it would need to be :shrug: 

> take care to use double ticks `` for inline code (single ticks are displayed as italic text)

Done.

noave commented on source/guide_mautic.rst:
```diff
@@ -0,0 +1,381 @@
+
+This guide will show you how to install Mautic 4 on your Uberspace using the Composer installation method.
+
+.. important:: It is bad practice to run your Mautic along your blog on the same server.
```
You can ditch this whole box, just use only "isabell.uber.space" as this is the standard procedure in all guides.

putzwasser replied:
Won't change. I don't see the point to simplify here. People use Mautic via a subdomain of their TLD. So in 99% of the use cases Mautic is set up to be reached via `(nl|newsletter|mautic).mysite.com`.

This needs a special configuration: suppressing some of the headers as explained in the guide below. This sets this guide apart from most of the other guides where a custom domain doesn't need special configuration.

Why would you make thing more complicated for users? The header thing took me quite a while to figure out. Ditching the subdomain from this guide certainly won't enhance the quality of the guide.

In the rare occasion some will use Mautic only by using their ubserspace subdomain they can simply ignore setting up the subdomain.

If you insist, please change it yourself.

noave replied:
I get your point and it is worthwhile to reference to the header, but it is separated from the domain setup and furthermore linked to the usage of forms, you can check how I did it now in my commits: https://deploy-preview-1506--uberlab.netlify.app/guide_mautic/#suppressing-security-header-for-external-origins

noave commented on source/guide_mautic.rst:
```diff
@@ -0,0 +1,381 @@
+  Project configuration is disabled: "symfony/flex" not found in the root composer.json
+
+  Congratulations, you’ve installed the Mautic codebase
+  from the mautic/recommended-project template!
```
reduce the output here a lot, just keep these two lines with "congratulations youve installed..." and set a `[...]` before and after

noave commented on source/guide_mautic.rst:
```diff
@@ -0,0 +1,381 @@
+----------------------------------
+
+Visit ``https://mautic.mysite.com`` (or ``https://isabell.uber.space``) in your browser.
+There might be a ``403 forbidden error``.
```
there wont be an error when following the guide, you can remove the whole part about the 403 error

noave commented on source/guide_mautic.rst:
```diff
@@ -0,0 +1,381 @@
+  3. You get your `bounces <https://docs.mautic.org/en/channels/emails/bounce-management>`_ from somewhere. If you don't or handle them somehow by yourself you can omit the line containing ``mautic:emails:fetch``.
+  4. You are subject to the European GDPR regulation. If you sure you aren't (*which is highly unlikely given it's architecture*) you could omit the the line containing ``mautic:maintenance:cleanup --days-old=365``. This simply removes data older than one year, which is probably advisable regardless of the GDPR.
+
+We save the logs of Mautic's console. If you don't want this remove the part after ``>>`` in your crontab.
```
this is good, maybe there is way to route also logs from the app (not only cronjobs) to this folder

putzwasser replied:
I don't know what you mean. The whole piping thing is about routing the logs to this folder.

noave replied:
there _might_ be even logs that are written by the app during normal usage (apart from cronjobs)

noave commented on source/guide_mautic.rst:
```diff
@@ -0,0 +1,381 @@
+
+.. code-block:: console
+
+  [isabell@stardust ~]$ cat << EOF
```
Dont do it like this. Just provide a codeblock (or multiple codeblocks, like described in my commend about the steps in line 280-282) with the cronjobs and tell the readers to add this to their cronjob. Add a link to the manual. See for example how its done here: https://lab.uberspace.de/guide_restic/#cronjob

putzwasser replied:
Done, but I don't feel that his enhances the quality of the guide. Instead of copying just one output, opening crontab, pasting the whole output and _maybe_ deleting some of the lines, people need to switch between editing crontab and the bash instance back and forth.

noave commented on source/guide_mautic.rst:
```diff
@@ -0,0 +1,381 @@
+.. important::
+  This guide assumes the following:
+
+  1. You're Mautic instance is installed in: ``/var/www/virtual/$USER/mautic``.
```
this is clear for the whole guide and you don't need to repeat it here

noave commented on source/guide_mautic.rst:
```diff
@@ -0,0 +1,381 @@
+  1. You're Mautic instance is installed in: ``/var/www/virtual/$USER/mautic``.
+  2. You use the `email queue <https://docs.mautic.org/en/contacts/message-queue>`_ (`Queue documentation <https://docs.mautic.org/en/queue>`_). If you don't you can omit the line containing ``mautic:emails:send``.
+  3. You get your `bounces <https://docs.mautic.org/en/channels/emails/bounce-management>`_ from somewhere. If you don't or handle them somehow by yourself you can omit the line containing ``mautic:emails:fetch``.
+  4. You are subject to the European GDPR regulation. If you sure you aren't (*which is highly unlikely given it's architecture*) you could omit the the line containing ``mautic:maintenance:cleanup --days-old=365``. This simply removes data older than one year, which is probably advisable regardless of the GDPR.
```
please move all this steps (2-4) to the cronjob part directly. provide the cronjob line for each step like "if you use the email queue add this cronjob:"

noave reviewed (changes requested):
I could successfully install the tool following the basics of this guide, but you can  leave out some parts and keep it more simple, see comments in text.

When you are ready, please set the PR back to "ready".

noave commented:
If you don't have the capacity to implement those changes its ok, we will take over after the next two weeks to complete everything, just for your information :-)

noave commented:
Ok thanks for your patience, I fixed up the rest I had in mind and will merge now :tada: please ping us at hallo@uberspace.de for a new cup if you like :tada: :coffee:
